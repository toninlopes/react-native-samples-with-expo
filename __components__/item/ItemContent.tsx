import React from "react";
import { Text, StyleSheet, View } from "react-native";
import type { ItemContentProps } from './types';

const ItemContent: React.FC<ItemContentProps> = ({ title }) => {
  return (
    <View style={styles.view}>
      <Text style={styles.title}>{title}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  view: {
    flex: 1,
  },
  title: {
      fontSize: 22,
  },
});

export default ItemContent;