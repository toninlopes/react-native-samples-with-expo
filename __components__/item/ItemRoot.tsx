import React from 'react';
import { TouchableOpacity, ViewProps } from 'react-native';
import { Link } from 'expo-router';
import type { ItemRootProps } from './types';

const ItemRoot: React.FC<ItemRootProps> = ({ href, ...rest }) => {

  if (href) {
    return (
      <Link asChild href={href}>
        <TouchableOpacity {...rest} />
      </Link>
    );  
  }

  return (
    <TouchableOpacity {...rest} />
  );
}

export default ItemRoot;