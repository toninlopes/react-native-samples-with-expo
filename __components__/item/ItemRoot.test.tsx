import React from 'react';
import renderer from 'react-test-renderer';
import ItemRoot from './ItemRoot';
import ItemContent from './ItemContent';

describe('ItemRoot', () => {
  test('When there is empty children.', () => {
    const itemRoot = renderer.create(<ItemRoot></ItemRoot>).toJSON();
    expect(itemRoot).toMatchSnapshot();
  });

  test('When there is one child.', () => {
    const itemRoot = renderer.create((
      <ItemRoot>
        <ItemContent title='Title' />
      </ItemRoot>
    )).toJSON();
    expect(itemRoot).toMatchSnapshot();
  });
});