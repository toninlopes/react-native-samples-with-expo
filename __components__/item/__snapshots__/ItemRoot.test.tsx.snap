// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemRoot When there is empty children. 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
/>
`;

exports[`ItemRoot When there is one child. 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 22,
        }
      }
    >
      Title
    </Text>
  </View>
</View>
`;
